include/exec/log_instr.h

```diff
@@ -129,6 +138,30 @@
 
 #ifdef CONFIG_TCG_LOG_INSTR
 
+/* Will generate TCG that will perform a printf to the extra logging info.
+ * Prints are batched until the end of an instruction
+ * TODO: This was actually designed to batch for an entire basic block, but
+ * sadly the string this would append to is
+ * TODO: cleared at the end of every instruction.
```

Comment:
This may yield different results when buffering is enabled. Maybe we can add a mode that buffers the logging for a basic block and automatically flush on exit, so that the buffer is retained for every instruction.